logbook/logbook.go

```diff
@@ -92,6 +93,7 @@ type Book struct {
 	pk         crypto.PrivKey
 	authorID   string
 	authorName string
+	bus        event.Bus
```

Comment:
I'm a fan of this. We may want to consider adding a `Publisher` interface to the `event` package that _only_ has the publish method, and make this a publisher field. Setup would be the same, just would add extra assurances that people can't subscribe in weird places.

We could also define a `NilPublisher` implementation that is nil-callable, which would drop the need for 
`if book.bus !=nil {` every time we want publish.

Reply:
Added both